src/mapping/parser/mod.rs

```diff
@@ -302,14 +308,55 @@ fn positional_item_from_pair(
     Ok(())
 }
 
+fn argument_item_from_pair(parameter: Parameter, pair: Pair<Rule>) -> Result<ArgumentKind> {
+    let inner = pair.into_inner().next().ok_or(TOKEN_ERR)?;
+    match inner.as_rule() {
+        Rule::query_arithmetic_boolean => query_arithmetic_from_pair(inner)
+            .map(|resolver| ArgumentKind::Value(Box::new(TypedArgument::new(resolver, parameter)))),
+        Rule::regex => regex_from_pair(inner).map(ArgumentKind::Regex),
+        _ => Err(TOKEN_ERR.into()),
+    }
+}
+
+fn regex_from_pair(pair: Pair<Rule>) -> Result<RemapRegex> {
+    match pair.as_rule() {
+        Rule::regex => {
+            let mut inner = pair.into_inner();
+            let regex = inner.next().ok_or(TOKEN_ERR)?;
+            let flags = inner.next().map(|flags| flags.as_str()).unwrap_or("");
+
+            let mut global = false;
+            let mut insensitive = false;
+            let mut multi_line = false;
+            for flag in flags.chars() {
+                if flag == 'i' {
+                    insensitive = true;
+                } else if flag == 'g' {
+                    global = true;
+                } else if flag == 'm' {
+                    multi_line = true;
+                }
```

Comment:
```suggestion
                match flag {
                    'i' => insensitive = true,
                    'g' => global = true,
                    'm' => multi_line = true,
                    _ => (),
                }
```

Reply:
Could also do this with `let (global, insensitive, multi_line) = flags.chars().fold(…)` but it's probably not worth it.